presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -286,7 +286,7 @@ protected RelationType visitShowTables(ShowTables showTables, AnalysisContext co
                 predicate,
                 ordering(ascending("table_name")));
 
-        analysis.setRowCountQuery(true);
+        analysis.setRowCountQuery(false);
```

Comment:
I propose we set this field to `false` by default and set it to `true` only in the row count queries (because they are fewer than the data set queries). What do you think?